### ISSUE TITLE
Error when computing automorphisms of simple orthogonal groups

### ISSUE THREAD

**Issue by savask** [gap-system/gap#4318]
### Observed behaviour

```
gap> G := PrimitiveGroup(119, 1);
O-(8, 2)
gap> S := SymmetricGroup(119);
Sym( [ 1 .. 119 ] )
gap> IsSubgroup(S, G);
true
gap> N := Normalizer(S, G);;
gap> StructureDescription(N);
"O-(8,2)"
gap> Index(N, G);
1
gap> T := TwoClosure(G);
<permutation group with 32 generators>
gap> StructureDescription(T);
"O-(8,2) : C2"
gap> Index(T, G);
2
```

### Expected behaviour

`G` is a subgroup of index 2 in `T`, so `T` must lie in the normalizer `N`.

`NormalizerPermGroup` outputs a correct answer:
```
gap> N2 := NormalizerPermGroup(S, G);;
gap> Index(N2, G);
2
```

### GAP banner
```
 ┌───────┐   GAP 4.10.0 of 01-Nov-2018
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64
 Configuration:  gmp 6.1.2, readline
 Loading the library and packages ...
 Packages:   Alnuth 3.1.0, AtlasRep 1.5.1, AutPGrp 1.10, CTblLib 1.2.2, FactInt 1.6.2, GAPDoc 1.6.2, IO 4.5.4, Polycyclic 2.14, PrimGrp 3.4.0, 
             SmallGrp 1.3, TomLib 1.2.7, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
```

The same problem is present in the current dev version 4.12dev-731-gc7a0c6e.

**Comment by hulpke** [comment]
[Dear "savask", Would you be willing to tell us your name and affiliation? I find it very awkward when working with people whose only identification is an anonymous handle. Thank you (Alexander Hulpke, Colorado State University)]

As for the bug, the issue ultimately is that GAP believes the group has no outer automorphisms. This stems from the table on the top of page xvi in the ATLAS, which calculates values d=1, f=1, g=1 and thus no automorphisms.
This is clearly wrong, but I need to find out what exactly the error is here (is the table wrong only for characteristic 2, or always) I will see to provide at least a workaround  for the issue.

**Comment by savask** [comment]
@hulpke I'm Saveliy Skresanov from Sobolev Institute of Mathematics (I'm more accustomed to using nicknames on the Internet, though).

Thank you for quick action. As for the Table 5 in the ATLAS, I have the following line:

![image](https://user-images.githubusercontent.com/343574/111260255-10f79280-8653-11eb-9113-ffb4e1092064.png)

It seems that n = 4 and q = 2 in our case, so the table gives f = 2.

**Comment by hulpke** [comment]
@savask Thank you very much -- I find it helpful to know a person's name, as there is a chance I would encounter it in real live in the future.

Indeed, I did stupidly misinterpret the q^2=p^f bit when I wrote the original code. Apologies!
 What happens is that the normalizer calculation stops in certain cases if the outer automorphisms indicate no further normalizing elements can exist.

I have checked the relevant code and the only groups affected by this bug will have been O_n(q) for n>=8 even. (so any calculations you did with other groups will be safe).  There is a fix for it on github, but please let me know if you'd like a patch file.

**Comment by wilfwilson** [comment]
Thank you for spotting and reporting this problem, @savask, and thanks for the fix in #4320 @hulpke. Hopefully this fix will be merged and included in the GAP 4.12 release in the near future.
